src/main/java/com/google/sps/placeGuide/repository/impl/DatastorePlaceGuideRepository.java

```diff
@@ -52,6 +61,30 @@ private Entity createPlaceGuideEntity(PlaceGuide placeGuide) {
     return placeGuideEntity;
   }
 
+  @Override
+  public void bookmarkPlaceGuide(long placeGuideId, String userId) {
+    // Check if the corresponding placeGuide is in the database.
+    Key placeGuideEntityKey = KeyFactory.createKey(ENTITY_KIND, placeGuideId);
+    try {
+      Entity placeGuideEntity = datastore.get(placeGuideEntityKey);
```

Comment:
As you don't actually need the placeGuideEntity, maybe you could implement a function which checks if a placeGudie exists or not in the database. Bookmark it only if it exists. 
It would be more readable then with these try-catch statements.

Reply:
done.